### PR TITLE
Added tests for input

### PR DESCRIPTION
**Pull request by amaldinesh7**
#947 

@ajmaln _a Please review

### CODE REVIEW

File: lib/components/Input.js [hunk 1]
```diff
@@ -66,7 +66,11 @@ const Input = forwardRef((props, ref) => {
           "neeto-ui-input--small": size === "small",
         })}
       >
-        {prefix && <div className="neeto-ui-input__prefix">{prefix}</div>}
+        {prefix && (
+          <div className="neeto-ui-input__prefix" data-testid="input_prefix">
```
Comment (ajmaln) [inline lib/components/Input.js:70]: Let's remove the testid from here, since it is not used in the test.

File: lib/components/Input.js [hunk 2]
```diff
@@ -79,12 +83,17 @@ const Input = forwardRef((props, ref) => {
...
-          data-cy={"input-field"}
+          data-cy="input-field"
           {...otherProps}
           value={value}
           onChange={onChange}
+          data-testid="input-field"
         />
-        {suffix && <div className="neeto-ui-input__suffix">{suffix}</div>}
+        {suffix && (
+          <div className="neeto-ui-input__suffix" data-testid="input_suffix">
```
Comment (ajmaln) [inline lib/components/Input.js:93]: Let's remove the testid from here, since it is not used in the test.

File: lib/components/Input.js [hunk 3]
```diff
@@ -79,12 +83,17 @@ const Input = forwardRef((props, ref) => {
             [errorId]: !!error,
             [helpTextId]: helpText,
           })}
-          data-cy={"input-field"}
+          data-cy="input-field"
           {...otherProps}
           value={value}
           onChange={onChange}
+          data-testid="input-field"
```
Comment (ajmaln) [inline lib/components/Input.js:90]: We dont need the testid here, since the input element can be found by the label.

File: tests/Input.test.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    const { getByLabelText, getByTestId } = render(
+      <Input label="Input Label" />
+    );
+    expect(getByLabelText("Input Label")).toBeInTheDocument();
+    expect(getByTestId("input-field")).toBeInTheDocument();
+  });
+
+  it("should be able to type when uncontrolled", () => {
+    const { getByTestId } = render(<Input label="input" />);
+    const inputField = getByTestId("input-field");
```
Comment (ajmaln) [inline tests/Input.test.js:17]: Lets use `getByLabelText` here

File: tests/Input.test.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    userEvent.type(inputField, "sample content");
+    expect(inputField).toHaveValue("sample content");
+  });
+
+  it("should call onChange when textarea value changes", () => {
+    const onChange = jest.fn();
+    const { getByTestId } = render(<Input label="input" onChange={onChange} />);
+    userEvent.type(getByTestId("input-field"), "Test");
```
Comment (ajmaln) [inline tests/Input.test.js:25]: Lets use `getByLabelText` here

**Review by ajmaln — COMMENTED** [review]
@amaldinesh7 _a Few changes. Also, let's add a test cover the `maxLength` prop.

image

**Comment by amaldinesh7** [comment]
@ajmaln _a Fixed the comments. Please check.

**Review by ajmaln — APPROVED** [review]

**Pull request merged**